git-config/main.tf

```diff
@@ -26,6 +26,17 @@ variable "allow_email_change" {
   default     = false
 }
 
+variable "coder_parameter_user_email_order" {
+  type        = number
+  description = "The order determines the position of the 'user_email' template parameter in the UI/CLI presentation. The lowest order is shown first and parameters with equal order are sorted by name (ascending order)."
+  default     = null
+}
+
+variable "coder_parameter_username_order" {
+  type        = number
+  description = "The order determines the position of the 'username' template parameter in the UI/CLI presentation. The lowest order is shown first and parameters with equal order are sorted by name (ascending order)."
+  default     = null
+}
```

Comment:
Would it be feasible to just use one `coder_parameter_order` for all parameters? If someone wants to change these individually, then there may be something wrong with the module or how it's being used.

Why? This makes modules updates more risky. If we ever add a third parameter, it could end up anywhere when the user updates.

Perhaps we can still apply order if it's given, e.g. `order = var.coder_parameter_order ? var.coder_parameter_order + 0 : null`, `+ 1`, etc.

An author can reserve as much ordered space for a module as they wish.

```tf
module "git-config" {
  source                = "registry.coder.com/modules/git-config/coder"
  version               = "1.0.11"
  agent_id              = coder_agent.example.id
  allow_email_change    = true
  coder_parameter_order = 10
}

module "gcp_region" {
  source                = "registry.coder.com/modules/gcp-region/coder"
  version               = "1.0.11"
  regions               = ["us", "europe"]
  coder_parameter_order = 20
}
```

To clarify this, maybe we could name it something like `coder_parameter_group_order` or `coder_parameter_order_weight`, I'm open to better ideas 😄.

Reply:
True, i can keep it all the same field name.

Reply:
@mafredri  - i have updated the code and tests